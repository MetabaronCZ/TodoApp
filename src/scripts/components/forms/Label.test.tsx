import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { Label } from 'components/forms/Label';

import { theme } from 'modules/theme';
import { hexToRgb } from 'modules/colors';
import { withMockedProviders } from 'test/component';

describe('components/forms/Label', () => {
  it('should render correctly', () => {
    const { container } = render(withMockedProviders(<Label text="Label" />));
    expect(container).toMatchSnapshot();
  });

  it('should render props', () => {
    const { container } = render(
      withMockedProviders(<Label text="Label" htmlFor="ID" disabled />),
    );
    expect(container).toMatchSnapshot();

    const label = container.querySelector('label');
    expect(label).toBeInTheDocument();
    expect(label).toHaveAttribute('for', 'ID');
    expect(label).toHaveStyle({ color: hexToRgb(theme.color.disabled) });
  });
});
