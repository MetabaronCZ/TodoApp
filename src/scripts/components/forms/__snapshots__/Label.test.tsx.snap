// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/Label should render correctly 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #222;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <label
    class="c0"
  >
    Label
  </label>
</div>
`;

exports[`components/forms/Label should render props 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <label
    class="c0"
    for="ID"
  >
    Label
  </label>
</div>
`;
